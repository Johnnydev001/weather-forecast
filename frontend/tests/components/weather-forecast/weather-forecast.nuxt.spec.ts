import { mountSuspended } from '@nuxt/test-utils/runtime';
import { it, describe, expect, vi} from 'vitest';
import WeatherForecastComponent from '~/components/weather-forecast/weather-forecast.vue';

describe('Weather forecast tests', () => {

    vi.mock('@/utils/utils', () => ({
        getImageUrlByWeatherStatus: vi.fn().mockReturnValue('/assets/imgs/cloudy.svg'),
        capitalizeWord: vi.fn().mockReturnValue('WORD')
    }))

    it('should match the snapshot', async () => {


        const props = [
            {
                dt: new Date().toString(),
                temp: {
                    min: 10,
                    max: 20
                },
                weather: [
                    {
                        main: 'cloudy',
                        description: 'Cloudy'
                    }
                ]
            }
        ]


        const weatherForecast = await mountSuspended(WeatherForecastComponent, {
            props: {
                weatherForecast: props
            }
        })

        expect(weatherForecast.html()).toMatchSnapshot()


    })

})
