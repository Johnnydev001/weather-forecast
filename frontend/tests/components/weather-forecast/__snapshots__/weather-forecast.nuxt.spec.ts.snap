// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Weather forecast tests > should match the snapshot 1`] = `
"<section data-v-83579346="" class="forecast-container">
  <h4 data-v-83579346="" class="title"> Next 5 days forecast: </h4>
  <ul data-v-83579346="" class="list">
    <li data-v-83579346="">
      <div data-v-7a63726d="" data-v-83579346="" class="forecast-item">
        <div data-v-7a63726d="" class="summary"><img data-v-7a63726d="" alt="Cloudy" data-nuxt-img="" srcset="/_ipx/_/assets/imgs/cloudy.svg 1x, /_ipx/_/assets/imgs/cloudy.svg 2x" class="weather-illustration" title="Cloudy" src="/_ipx/_/assets/imgs/cloudy.svg">
          <div data-v-7a63726d="" style="display: flex; column-gap: 0.5rem; align-items: center;"><span data-v-7a63726d="">Day:</span><time data-v-7a63726d="" datetime="Invalid Date">NaN</time></div><span data-v-7a63726d="">WORD</span>
        </div>
        <div data-v-7a63726d="" class="condition-container">
          <div data-v-7a63726d="" class="condition"><span data-v-7a63726d="">Min:</span>
            <div data-v-7a63726d="" class="value"><span data-v-7a63726d="" style="color: #ffffff; display: flex; column-gap: 0.25rem; font-weight: 800;"><span data-v-7a63726d="">10</span><span data-v-7a63726d="">ºC</span></span></div>
          </div>
          <div data-v-7a63726d="" class="condition"><span data-v-7a63726d="">Max:</span>
            <div data-v-7a63726d="" class="value"><span data-v-7a63726d="" style="color: orange; display: flex; column-gap: 0.25rem; font-weight: 800;"><span data-v-7a63726d="">20</span><span data-v-7a63726d="">ºC</span></span></div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</section>"
`;
